lms/djangoapps/courseware/module_render.py

```diff
@@ -157,6 +157,7 @@ def toc_for_course(request, course, active_chapter, active_section, field_data_c
         for chapter in chapters:
             # Only show required content, if there is required content
             # chapter.hide_from_toc is read-only (boo)
+            name_without_spaces = chapter.display_name_with_default.replace(" ", "-").replace(":", "")
```

Comment:
There are all kinds of other characters that might cause problems, so I recommend using a more comprehensive solution. The T&L team is introducing a ``slugify`` function from Django 1.8 that I would recommend. Unfortunately it hasn't been merged yet, but you could add it in your PR too (or wait for https://github.com/edx/edx-platform/pull/8243 to land):

https://github.com/edx/edx-platform/blob/benmcmorran/team-api-a/common/djangoapps/util/model_utils.py#L154